dbms/src/Storages/DeltaMerge/Index/IndexInfo.cpp

```diff
@@ -0,0 +1,208 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Storages/DeltaMerge/Index/IndexInfo.h>
+#include <Storages/FormatVersion.h>
+#include <Storages/KVStore/Types.h>
+#include <TiDB/Schema/TiDB.h>
+
+#include <cstddef>
+
+namespace DB::DM
+{
+
+bool isVectorIndexSupported(const LoggerPtr & logger)
+{
+    // Vector Index requires a specific storage format to work.
+    if ((STORAGE_FORMAT_CURRENT.identifier > 0 && STORAGE_FORMAT_CURRENT.identifier < 6)
+        || STORAGE_FORMAT_CURRENT.identifier == 100)
+    {
+        LOG_ERROR(
+            logger,
+            "The current storage format is {}, which does not support building vector index. TiFlash will "
+            "write data without vector index.",
+            STORAGE_FORMAT_CURRENT.identifier);
+        return false;
+    }
+
+    return true;
+}
+
+TiDB::ColumnInfo getVectorIndxColumnInfo(
+    const TiDB::TableInfo & table_info,
+    const TiDB::IndexInfo & idx_info,
+    const LoggerPtr & logger)
+{
+    if (!idx_info.vector_index
+        || (idx_info.state != TiDB::StatePublic && idx_info.state != TiDB::StateWriteReorganization))
+    {
+        return {};
+    }
+
+    // Vector Index requires a specific storage format to work.
+    if (!isVectorIndexSupported(logger))
+    {
+        return {};
+    }
+
+    if (idx_info.idx_cols.size() != 1)
+    {
+        LOG_ERROR(
+            logger,
+            "The index columns length is {}, which does not support building vector index, index_id={}, table_id={}.",
+            idx_info.idx_cols.size(),
+            idx_info.id,
+            table_info.id);
+        return {};
+    }
+
+    for (const auto & col : table_info.columns)
+    {
+        if (col.name == idx_info.idx_cols[0].name)
+        {
+            return col;
+        }
+    }
+
+    LOG_ERROR(
+        logger,
+        "The index column does not exist, table_id={} index_id={} idx_col_name={}.",
+        table_info.id,
+        idx_info.id,
+        idx_info.idx_cols[0].name);
+    return {};
+}
+
+LocalIndexInfosPtr initLocalIndexInfos(const TiDB::TableInfo & table_info, const LoggerPtr & logger)
+{
+    LocalIndexInfosPtr index_infos = std::make_shared<LocalIndexInfos>();
+    index_infos->reserve(table_info.columns.size() + table_info.index_infos.size());
+    for (const auto & col : table_info.columns)
```

Comment:
We needn't use this logic, because we don't support adding a vector index with a column comment.

Reply:
The later PR merge the logic of function `initLocalIndexInfos` into `generateLocalIndexInfos`. I think we can remove this for-loop after all PRs from the cse branch are merged? Or this could introduce unnecessary conflict when cherry-pick later PRs

Reply:
Got it.